src/AI/AIUtils.java

```diff
@@ -63,7 +63,7 @@
   {
     // Create the ActionType-indexed map, and ensure we don't have any null pointers.
     Map<UnitActionType, ArrayList<GameAction> > actionsByType = new HashMap<UnitActionType, ArrayList<GameAction> >();
-    for( UnitActionType atype : UnitActionType.GENERIC_ACTIONS )
+    for( UnitActionType atype : unit.model.possibleActions )
```

Comment:
Are you sure this change is safe? AIs just assume that, for example, ATTACK is always populated. If one ended up with an APC or T-Copter, they might crash due to pulling a null array from this function's result map.

Reply:
Then they had better not make assumptions. I think it's perfectly valid for `getUnitActionsByType` to return a map that doesn't contain entries for unsupported action types.

Reply:
Then I suppose you'd better fix it?
Or I guess I could?
*shrug*

Reply:
I can see the convenience of guaranteeing that the standard action types will be represented even if the collection is empty, but that also makes it so you have to check for null under certain circumstances, but not others, and I don't like making the interface inconsistent.

Side note: If we do change it back, the second part of this function will break if Muriel tries to move a Meridian tank, because the call to `getAvailableUnitActions` will return a type that doesn't already exist in `actionsByType`.

Reply:
Yeah. That's basically what was happening with the save/load with AI bug we discussed.
I don't care too much which way we go, so long as we don't crash or anything.